### ISSUE TITLE
Errors while building using amalgamation 

### ISSUE THREAD

**Issue by ZigZag-Builds** [Zeex/sampgdk#172]
Hi there,

I know in the discussion here https://github.com/Zeex/sampgdk/issues/90 this problem already appeared but it did not help. 
So I used your tutorial https://github.com/Zeex/sampgdk/wiki/Setting-up-GDK-with-CMake. I followed every step. When I used CMake I decided to use the Visual Studio 2015 compiler.

After I built it, I opened the project solution. I did nothing to it. And I pressed "build solution". Then I got these errors 
`Error	C2373	'sampgdk_GetTickCount': redefinition; different type modifiers	helloworld	C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\um\sysinfoapi.h	210	
Error	C2373	'sampgdk_SelectObject': redefinition; different type modifiers	helloworld	C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\um\wingdi.h	4473	
Error	C2373	'sampgdk_SetTimer': redefinition; different type modifiers	helloworld	C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\um\winuser.h	6803	
Error	C2373	'sampgdk_CreateMenu': redefinition; different type modifiers	helloworld	C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\um\winuser.h	7273	
Error	C2373	'sampgdk_DestroyMenu': redefinition; different type modifiers	helloworld	C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\um\winuser.h	7285	
`
In your earlier discussion (which I already mentioned) you said, that it could be that it sampgdk.h was included before windows.h.  
Since I just used your hello world example with these includes:
`#include <stdio.h>
#include <string.h>

#include "sampgdk.h"`
I don't think that there is a problem.

Another proposal was to add a new definition which I think is covered with this cmake line:
`add_definitions(-DSAMPGDK_AMALGAMATION)`

I also tried to do it anyway with your instructions:
"In Visual Studio you would do it like this:
Right click on your Projrect -> Properties -> C/C++ -> Preprocessor -> Preprocessor Definitions -> Edit"
but I saw that the definition was already added.

So I wanted to know if I did anything wrong or whether you have some approaches to solve the problem.

**Comment by WopsS** [comment]
Hello,

Add this
```cpp
#define IN_SAMPGDK

#ifndef SAMPGDK_AMALGAMATION
  #define SAMPGDK_AMALGAMATION
#endif

#include "sampgdk.h"

#if SAMPGDK_WINDOWS
  #ifdef _MSC_VER
    #pragma warning(disable: 4996)
  #endif
  #undef CreateMenu
  #undef DestroyMenu
  #undef GetTickCount
  #undef KillTimer
  #undef SelectObject
  #undef SetTimer
  #define WIN32_LEAN_AND_MEAN
  #include <windows.h>
#endif
```

at the top of **sampgdk.c**. I don't know why it wasn't included in **v4.5.2**, but in the v4.5.1 or v4.5 it is.

**Comment by Zeex** [comment]
Don't know why that code was missing, probably something with the CI build. I re-uploaded the amalgamation zip, it should be ok now.
